modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLIdPInitiatedSSOTestCase.java

```diff
@@ -0,0 +1,173 @@
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initTest() throws Exception {
+
+        super.init();
+        super.testInit();
+        addServiceProvider();
+        httpClient = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).disableRedirectHandling().build();
+        super.createUser(samlConfig);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+
+        super.deleteUser(samlConfig);
+        super.ssoConfigServiceClient.removeServiceProvider(SP_NAME);
+        super.deleteApplication(SP_NAME);
+        super.testClear();
+    }
+
+    @Test(groups = "wso2.is", description = "This test method will test the IdP initiated SAML SSO process.")
+    public void testIdPInitiatedSSO() throws Exception {
+
+        HttpResponse idpInitResponse =
+                Utils.sendGetRequest(String.format(IDP_INIT_SSO_URL, IS_DEFAULT_HTTPS_PORT), USER_AGENT,
+                        httpClient);
+        String redirectUrl = Utils.getRedirectUrl(idpInitResponse);
+        Assert.assertTrue(redirectUrl.contains("/authenticationendpoint/login.do"), "Cannot find the login page.");
+
+        String sessionDataKey = getSessionDataKeyFromRedirectUrl(redirectUrl);
+
+        HttpResponse samlssoResponse = Utils.sendPOSTMessage(sessionDataKey, String.format(SAML_SSO_URL,
+                IS_DEFAULT_HTTPS_PORT), USER_AGENT, SP_ACS_URL, samlConfig.getApp()
+                        .getArtifact(), samlConfig.getUser().getUsername(), samlConfig.getUser().getPassword(),
+                httpClient);
+
+        List<NameValuePair> consentRequiredClaims = Utils.getConsentRequiredClaimsFromResponse(samlssoResponse);
+        HttpResponse commonAuthResponse = setConsentForSP(sessionDataKey, consentRequiredClaims);
+
+        String samlRedirectUrl = Utils.getRedirectUrl(commonAuthResponse);
+        HttpResponse samlRedirectResponse = Utils.sendGetRequest(samlRedirectUrl, USER_AGENT, httpClient);
+
+        String samlRedirectPage = extractDataFromResponse(samlRedirectResponse);
+        Assert.assertTrue(samlRedirectPage.contains(String.format(SP_ACS_URL, SP_NAME)), "Cannot find the assertion " +
+                "consumer URL in the resulting page.");
+
+    }
+
+    private void addServiceProvider() throws Exception {
+
+        super.createApplication(samlConfig, SP_NAME);
+        super.ssoConfigServiceClient.addServiceProvider(super.createSsoSPDTOForIdPInit(samlConfig));
+        log.info("Service Provider " + samlConfig.getApp().getArtifact() + " created.");
+
+    }
+
+    private String getSessionDataKeyFromRedirectUrl(String redirectUrl) throws Exception {
+
+        String sessionDataKey = "";
+        Map<String, String> queryParams = Utils.getQueryParams(redirectUrl);
+        for (Map.Entry<String, String> paramEntry : queryParams.entrySet()) {
+            if (paramEntry.getKey().equals("sessionDataKey")) {
```

Comment:
Can we please swap the constant with the variable to avoid possible null pointers

Reply:
Done.